rules/Renaming/Rector/Class_/RenameAttributeRector.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Renaming\Rector\Class_;
+
+use PhpParser\Node;
+use PhpParser\Node\Name\FullyQualified;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Property;
+use Rector\Contract\Rector\ConfigurableRectorInterface;
+use Rector\Rector\AbstractRector;
+use Rector\Renaming\ValueObject\RenameAttribute;
+use Rector\ValueObject\PhpVersionFeature;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\ConfiguredCodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+use Webmozart\Assert\Assert;
+
+/**
+ * @see \Rector\Tests\Renaming\Rector\Class_\RenameAttributeRector\RenameAttributeRectorTest
+ */
+final class RenameAttributeRector extends AbstractRector implements ConfigurableRectorInterface, MinPhpVersionInterface
+{
+    /**
+     * @var RenameAttribute[]
+     */
+    private array $renameAttributes = [];
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Rename attribute class names',
+            [
+                new ConfiguredCodeSample(
+                    <<<'CODE_SAMPLE'
+#[SimpleRoute()]
+class SomeClass
+{
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+#[BasicRoute()]
+class SomeClass
+{
+}
+CODE_SAMPLE
+                    ,
+                    []
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class, ClassMethod::class, Property::class];
+    }
+
+    /**
+     * @param Class_|ClassMethod|Property $node
+     */
+    public function refactor(Node $node): ?\PhpParser\Node
+    {
+        $hasChanged = false;
+
+        foreach ($node->attrGroups as $attrGroup) {
+            foreach ($attrGroup->attrs as $attr) {
+                foreach ($this->renameAttributes as $renameAttribute) {
+                    if ($this->isName($attr->name, $renameAttribute->getOldAttribute())) {
+                        $attr->name = new FullyQualified($renameAttribute->getNewAttribute());
+
+                        $hasChanged = true;
```

Comment:
Here needs `continue 2`, if already renamed, continue to next attr

Reply:
Got it, thanks :+1: